### PR TITLE
Added GitHub Actions Workflow

### PR DESCRIPTION
**Pull request by Lemeri123**
Added .github/workflows/main.yml
   Configured to run on push and pull_request events.
   Utilizes the gradlew wrapper for build and test tasks.
Testing
    Verified that all tests pass locally after the fix.
    Confirmed that the new workflow triggers successfully in GitHub Actions.
    Ensured compatibility with the existing CI/CD pipeline.

### CODE REVIEW

**Comment by Lemeri123** [comment]
@josdem please review this and let me know if I should make any changes

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name: CI - Test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
```
Comment (josdem) [inline .github/workflows/main.yml:7]: Please add `pull_request` on branches: `main` too

**Review by josdem — CHANGES_REQUESTED** [review]
Hi @Lemeri123 , thank you for sending a PR, just a small change before merge.

**Comment by josdem** [comment]
@all-contributors please add @Lemeri123 for infra

**Review by josdem — APPROVED** [review]
Thanks for doing suggested changes 👍

**Pull request merged**